### PR TITLE
feature: official/tasks: check for valid contact info after USVF sync

### PR DESCRIPTION
**Pull request by liewegas**
- usvf sync runs the check after the sync.
- check_vbm_states() is a separate task so it can be run independently
- check_state_contacts takes the vbm_submission_type, so that a state can
be check for a mode that is not currently set (to see whether the change is
safe to make).

### CODE REVIEW

**Comment by liewegas** [comment]
This can be run manually with something like
``
make shell
echo -e 'from common import enums\nimport official.tasks\nprint(official.tasks.check_state_contacts("WI", enums.SubmissionType.LEO_FAX))' | ./manage.py shell
``

File: app/official/tasks.py [hunk 1]
```diff
@@ -1,8 +1,101 @@
...
 
 @shared_task
 def sync_usvotefoundation():
     from .usvf_sync import sync
 
     sync()
+    check_vbm_states()
+
+
+def check_state_contacts(state_id: str, mode: enums.SubmissionType) -> str:
```
Comment (nickcatal) [inline app/official/tasks.py:26]: Can we move this out of `tasks.py`?

File: app/official/tasks.py [hunk 2]
```diff
@@ -1,8 +1,101 @@
...
+    return None
+
+
+@shared_task
+def check_vbm_states():
+    for state in State.objects.all():
+        if state.vbm_submission_type != "self_print":
+            message = check_state_contacts(state.code, state.vbm_submission_type)
+            if message:
+                logger.warning(message)
```
Comment (nickcatal) [inline app/official/tasks.py:93]: I believe this is going to pipe the error to Sentry. Do we want this behavior?

File: app/official/tasks.py [hunk 3]
```diff
@@ -1,8 +1,101 @@
...
+                if settings.SLACK_DATA_ERROR_ENABLED:
+                    try:
+                        r = requests.post(
+                            settings.SLACK_DATA_ERROR_WEBHOOK, json={"text": message}
+                        )
```
Comment (nickcatal) [inline app/official/tasks.py:98]: Can we pipe monitoring of this to statsd?

File: app/official/tasks.py [hunk 4]
```diff
@@ -1,8 +1,101 @@
...
+
+logger = logging.getLogger("official")
 
 
 @shared_task
 def sync_usvotefoundation():
     from .usvf_sync import sync
 
     sync()
+    check_vbm_states()
```
Comment (nickcatal) [inline app/official/tasks.py:23]: Do we want to run this with `.delay()` or run it inline in the task?

File: app/official/tasks.py (hunk 1 above)
Comment (liewegas) [reply app/official/tasks.py:26]: where should this go?  i'm assuming not usvf_sync.py.  new file?  (validate.py?)

File: app/official/tasks.py (hunk 2 above)
Comment (liewegas) [reply app/official/tasks.py:93]: I think so?  If this triggers it's the live db that has the problem and it needs quick attention (vs `check_state_contacts()` which can be run manually for any state + mode)

File: app/official/tasks.py (hunk 3 above)
Comment (liewegas) [reply app/official/tasks.py:98]: do you mean something like
```
statsd.increment("turnout.official.missing_leo_contact_info")
```
?

File: app/official/tasks.py (hunk 4 above)
Comment (liewegas) [reply app/official/tasks.py:23]: Inline seems simpler.  The check is fast (esp compared to the sync).

File: app/official/tasks.py (hunk 1 above)
Comment (nickcatal) [reply app/official/tasks.py:26]: `usvf_validate.py`?

File: app/official/tasks.py (hunk 4 above)
Comment (nickcatal) [reply app/official/tasks.py:23]: Then is there a reason to leave it in `tasks.py`? Might make sense to throw in `usvf_validate.py`

File: app/official/tasks.py (hunk 2 above)
Comment (nickcatal) [reply app/official/tasks.py:93]: Makes sense to me. let's see if it gets annoying or not

File: app/official/tasks.py (hunk 1 above)
Comment (liewegas) [reply app/official/tasks.py:26]: i could merge it all into usvf.py..

Comment (nickcatal) [reply app/official/tasks.py:26]: Not a bad idea at all

**Pull request merged**